test/test_autograd.py

```diff
@@ -2634,15 +2634,23 @@ def test_record_function(self):
         with profile() as p:
             with record_function("label"):
                 y = x * 2 + 4
-
-        last_end = 0
-        names = ['mul', 'add']
-        labels = ['label']
-        self.assertEqual(len(p.function_events), len(names) + len(labels))
+            y = y + 1
+
+        last_start = 0
+        labels = [
+            'profiler::record_function_enter',
+            'label',
+            'mul',
+            'add',
+            'profiler::record_function_exit',
+            'add',
+        ]
+        print(p.function_events)
+        self.assertEqual(len(p.function_events), len(labels))
         for info, expected_name in zip(p.function_events, labels):
-            self.assertGreater(info.cpu_interval.start, last_end)
+            self.assertGreater(info.cpu_interval.start, last_start)
             self.assertEqual(info.name, expected_name)
-            last_end = info.cpu_interval.end
+            last_start = info.cpu_interval.start
```

Comment:
Shall we also check "label" event's time range is greater than "mul" and "add"?

Reply:
added a check for the end ordering